platform/error/platform-error-page-injector/src/main/java/org/codice/ddf/platform/error/injector/ErrorPageInjector.java

```diff
@@ -0,0 +1,204 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.platform.error.injector;
+
+import com.google.common.collect.ImmutableMap;
+import java.lang.reflect.Field;
+import java.util.Collection;
+import java.util.Map;
+import java.util.Optional;
+import javax.servlet.ServletContext;
+import org.codice.ddf.platform.error.servlet.ErrorServlet;
+import org.eclipse.jetty.servlet.ErrorPageErrorHandler;
+import org.eclipse.jetty.servlet.ServletContextHandler;
+import org.eclipse.jetty.servlet.ServletHandler;
+import org.eclipse.jetty.servlet.ServletHolder;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.Constants;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.InvalidSyntaxException;
+import org.osgi.framework.ServiceEvent;
+import org.osgi.framework.ServiceReference;
+import org.osgi.framework.hooks.service.EventListenerHook;
+import org.osgi.framework.hooks.service.ListenerHook.ListenerInfo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ErrorPageInjector implements EventListenerHook {
+
+  private final String errorPagePath = "/ErrorServlet";
+  private ImmutableMap<String, String> errorCodesMap =
```

Comment:
While I do like how that looks cleaner I also want to keep the map immutable without having to copy it.

Reply:
`Collections.unmodifiableMap` decorates map that can't be modified. Could either do it with
```java
    Map<String, String> foo =
        Collections.unmodifiableMap(Stream.of("400", "401", "403", "404", "405", "406", "500", "501", "502", "503", "504")
            .collect(Collectors.toMap(Function.identity(), x -> errorPagePath)));
```
or the slightly more elegant
```java
    Map<String, String> foo =
        Stream.of("400", "401", "403", "404", "405", "406", "500", "501", "502", "503", "504")
            .collect(
                Collectors.collectingAndThen(
                    Collectors.toMap(Function.identity(), x -> errorPagePath),
                    Collections::unmodifiableMap));
```
These are options, of course. The existing code it fine as-is.

Reply:
If you want immutability and performance - persistent data structures would be a good fit 😀

Reply:
This is a better solution in the long run I think as if someone ever wants to put the error codes in a list somewhere it will work better with a stream than what was originally there.